src/app/users/users.service.ts

```diff
@@ -48,6 +49,36 @@ export class UsersService {
     return this.http.get(`/users/${userId}`);
   }
 
+  /**
+   * Confirm Change Password of Users
+   * @param controlNameToCompare
+   */
+  confirmPassword(controlNameToCompare: string): ValidatorFn {
+    return (c: AbstractControl): ValidationErrors|null => {
+        if (c.value == null || c.value.length === 0) {
+            return null;
+        }
+        const controlToCompare = c.root.get(controlNameToCompare);
+        if (controlToCompare) {
+            const subscription: Subscription = controlToCompare.valueChanges.subscribe(() => {
+                c.updateValueAndValidity();
+                subscription.unsubscribe();
+            });
+        }
+        return controlToCompare && controlToCompare.value !== c.value ? {'notequal': true} : null;
+    };
+  }
```

Comment:
why is this in service? move it to dialog component.ts, also what did you refer to while writing code for this?

Reply:
Tried to compare passwor and re-enter password features. Can I move this to component.ts file?

Reply:
Added to the component.ts file

Reply:
sorry I forgot to delete this comment we will be having a password service containing these 2 functions

Reply:
keep password service in same folder as this dialog and keep the dialog in shared folder

Reply:
Added a new service file. Please review